packages/ckeditor5-dev-env/tests/release-tools/utils/generatechangelogfromcommits-integration.js

```diff
@@ -0,0 +1,136 @@
+			}
+
+			const packageJson = {
+				name: '@ckeditor/ckeditor5-test-package',
+				bugs: 'https://github.com/ckeditor/ckeditor5-test-package/issues',
+				repository: 'https://github.com/ckeditor/ckeditor5-test-package'
+			};
+
+			fs.writeFileSync(
+				path.join( tmpCwd, 'package.json' ),
+				JSON.stringify( packageJson, null, '\t' )
+			);
+		} );
+
+		after( () => {
+			exec( `rm -rf ${ tmpCwd }` );
+			process.chdir( cwd );
+		} );
+
+		it( 'generates a changelog for the first time', () => {
+			exec( 'git commit --allow-empty --message "Internal: An initial commit."' );
+
+			return generateChangelog( '0.0.1' )
+				.then( () => {
+					const changelog = getChangelog();
+
+					expect( changelog ).to.contain( changelogHeader );
+					expect( changelog ).to.contain( 'Internal changes only (updated dependencies, documentation, etc.).' );
+
+					release( '0.0.1' );
+				} );
+		} );
+
+		it( 'title of the next release should be a link which compares current version with the previous one', () => {
+			exec( 'git commit --allow-empty --message "Feature: Some amazing feature. Closes #1."' );
+
+			return generateChangelog( '0.1.0', '0.0.1' )
+				.then( () => {
+					const expectedTitle = '## [0.1.0](https://github.com/ckeditor/ckeditor5-test-package/compare/v0.0.1...v0.1.0)';
+
+					expect( getChangelog() ).to.contain( expectedTitle );
+
+					release( '0.1.0' );
+				} );
+		} );
+
+		it( 'does not hoist issues from the commit body', () => {
+			exec( 'git commit --allow-empty ' +
+				'--message "Feature: Another feature. Closes #2." ' +
+				'--message "This PR also closes #3 and #4."' );
+
+			return generateChangelog( '0.2.0', '0.1.0' )
+				.then( () => {
+					const changelog = getChangelog();
+					const url = 'https://github.com/ckeditor/ckeditor5-test-package';
+
+					//jscs:disable maximumLineLength
+					expect( changelog ).to.match( new RegExp( `\\* Another feature. Closes \\[#2\\]\\(${ url }\\/issues\\/2\\). \\(\\[[a-z0-9]{7}\\]\\(${ url }\\/commit\\/[a-z0-9]{7}\\)\\)` ) );
+					expect( changelog ).to.match( new RegExp( `  This PR also closes \\[#3\\]\\(${ url }\\/issues\\/3\\) and \\[#4\\]\\(${ url }\\/issues\\/4\\).` ) );
+					//jscs:enable maximumLineLength
+
+					release( '0.2.0' );
+				} );
+		} );
+
+		it( 'does not hoist issues from the commit body for merge commit', () => {
+			exec( 'git commit --allow-empty ' +
+				'--message "Merge pull request #5 from ckeditor/t/4" ' +
+				'--message "Fix: Amazing fix. Closes #5." ' +
+				'--message "The PR also finally closes #3 and #4. So good!"' );
+
+			return generateChangelog( '0.2.1', '0.2.0' )
+				.then( () => {
+					const changelog = getChangelog();
+					const url = 'https://github.com/ckeditor/ckeditor5-test-package';
+
+					//jscs:disable maximumLineLength
+					expect( changelog ).to.match( new RegExp( `\\* Amazing fix. Closes \\[#5\\]\\(${ url }\\/issues\\/5\\). \\(\\[[a-z0-9]{7}\\]\\(${ url }\\/commit\\/[a-z0-9]{7}\\)\\)` ) );
```

Comment:
In other words – use simple operations. Instead of having one huge regexp (which I cannot read) use smaller patterns to "unify" the result and then compare to a string.